src/media-groups.js

```diff
@@ -405,18 +405,18 @@ export const initialize = {
       for (const variantLabel in mediaGroups[type][groupId]) {
         let properties = mediaGroups[type][groupId][variantLabel];
 
-        // List of playlists for the current group ID that have a matching uri with
-        // this alternate audio variant
-        const matchingPlaylists = groupPlaylists.filter(playlist => {
-          return playlist.resolvedUri === properties.resolvedUri;
+        // List of playlists for the current group ID that do not have a matching uri
+        // with this alternate audio variant
+        const unmatchingPlaylists = groupPlaylists.filter(playlist => {
+          return playlist.resolvedUri !== properties.resolvedUri;
         });
 
-        if (matchingPlaylists.length) {
-          // If there is a playlist that has the same uri as this audio variant, assume
-          // that the playlist is audio only. We delete the resolvedUri property here
-          // to prevent a playlist loader from being created so that we don't have
-          // both the main and audio segment loaders loading the same audio segments
-          // from the same playlist.
+        // If there are no playlists using this audio group other than ones
+        // that match it's uri. Then playlist is audio only. We delete the resolvedUri
```

Comment:
One thing I'm wondering is: what if we have a manifest where there are two playlists using the audio group, one is audio only, and another uses it as its demuxed audio for video. Wouldn't we need the audio loader running for that case?

Reply:
We would but we don't support going from video/audio to just audio or vice versa, if I understand what you are saying.